services/queue/src/api.js

```diff
@@ -2251,14 +2251,22 @@ builder.declare({
   ].join('\n'),
 }, async function(req, res) {
   const { provisionerId, workerType, workerGroup, workerId } = req.params;
-  const { quarantineUntil } = req.body;
+  const { quarantineUntil, quarantineInfo } = req.body;
   const taskQueueId = joinTaskQueueId(provisionerId, workerType);
 
-  const [result] = await this.db.fns.quarantine_queue_worker_with_last_date_active({
+  const quarantineDetails = {
+    clientId: await req.clientId() || 'unknown-client',
```

Comment:
Oh! great catch.
It looks like middleware would give same string if it cannot determine clientId, so this check is not needed

https://github.com/taskcluster/taskcluster/blob/main/libraries/api/src/middleware/auth.js#L205-L212

Reply:
And I'm not completely sure where `clientId` is coming from. Most likely from the hawk headers. And most likely it would be set if that endpoint is accessible to anonymous scope